airflow/jobs/scheduler_job.py

```diff
@@ -970,48 +971,61 @@ def _start_queued_dagruns(
         session: Session,
     ) -> int:
         """Find DagRuns in queued state and decide moving them to running state"""
-        dag_runs = self._get_next_dagruns_to_examine(State.QUEUED, session)
-
-        active_runs_of_dags = defaultdict(
-            lambda: 0,
-            session.query(DagRun.dag_id, func.count('*'))
-            .filter(  # We use `list` here because SQLA doesn't accept a set
-                # We use set to avoid duplicate dag_ids
-                DagRun.dag_id.in_(list({dr.dag_id for dr in dag_runs})),
-                DagRun.state == State.RUNNING,
-            )
-            .group_by(DagRun.dag_id)
-            .all(),
+        active_dags = (
+            session.query(DM)
+            .filter(DM.is_active == expression.true(), DM.is_paused == expression.false())
+            .all()
         )
+        max_number = conf.getint('scheduler', 'max_dagruns_per_loop_to_schedule', fallback=20)
+        # There's a possibility of exceeding this max_number or going below it as a result of the logic below
+        if len(active_dags) > max_number:
+            max_number = 1
+        elif len(active_dags) > 0:
```

Comment:
For example, the total dags are 30, which is greater than `max_dagruns_per_loop_to_schedule`(20), so we check 1 dagrun from each dag which will now exceed the `max_dagruns_per_loop_to_schedule` at the end

Reply:
Another thing I have thought about is to allow it to use the `max_dagruns_per_loop_to_schedule` on each dag since it's only to  update the `dagrun` state at each loop and nothing more.